corehq/apps/api/resources/v0_5.py

```diff
@@ -1093,16 +1093,17 @@ def _query(cls, domain: str, local_time_zone: ZoneInfo, params: dict = {}):
 
         queryset = NavigationEventAudit.objects.filter(filters)
 
-        results = (queryset
-                .annotate(local_date=TruncDate('event_date', tzinfo=local_time_zone))
-                .values("local_date", "user"))
-
-        date_filters = Q()
+        date_filter = Q()
         if 'local_start_date' in params:
             date_filters &= Q(local_date__gte=params['local_start_date'])
         if 'local_end_date' in params:
             date_filters &= Q(local_date__lte=params['local_end_date'])
 
-        results = results.filter(date_filters).order_by('local_date', 'user')
+        results = (queryset
+                .annotate(local_date=TruncDate('event_date', tzinfo=local_time_zone))
+                .filter(date_filter)
+                .values("local_date", "user"))
```

Comment:
No, the `.filter(date_filter)` filters by the annotated local_date so it needs to go after. I originally thought it would help performance by filtering before grouping by values but I think SQL reorders the Django queries anyways, so it doesn't really mater.

Going to update variable to ` .filter(local_date_filter)` for clarity

Reply:
[4e034f8](https://github.com/dimagi/commcare-hq/pull/32922/commits/4e034f8c220883f3c4b4ed25918c524471990c2d)